app/helpers/pipeline_outputs_helper.rb

```diff
@@ -1,6 +1,24 @@
 module PipelineOutputsHelper
   Client = Aws::S3::Client.new
 
+  def curate_pipeline_run_display(pipeline_run)
+    return nil unless pipeline_run
+    pipeline_run_display = pipeline_run.as_json.except("version")
+    pipeline_run_display["version"] = { pipeline: select_version_aspect(pipeline_run, "idseq-pipeline"),
+                                        nt: select_version_aspect(pipeline_run, "nt_k16"),
+                                        nr: select_version_aspect(pipeline_run, "nr_rapsearch") }
+    pipeline_run_display
+  end
+
+  def select_version_aspect(pipeline_run, aspect)
+    return nil unless pipeline_run.version
+    version_hash = JSON.parse(pipeline_run.version)
+    aspect_hash = version_hash.select { |item| item["name"] == aspect }[0]
```

Comment:
Is the `[0]` guaranteed not to go out-of-bounds?

Reply:
`[0]` will just give nil if it is out-of-bounds and that case is then handled in the next line.
Ok, I'm adding an example of version_hash. It should actually be called version_array.